### PR TITLE
Fixes silo logging + add log file based logging to silo

### PR DESCRIPTION
**Pull request by Gamer025**
## About The Pull Request
Fixes deposit logs not working for the silo because they passed in the mats argument in the wrong format.
Also adds log file based logging to the silo with a new log file: silo.log.
This log should contain every deposit and withdraw action performed on the silo and from which machine and location inluding material amounts.
## Why It's Good For The Game
Fixes broken ingame logs + adds better external logs for potential stat tracking.
## Changelog
:cl: Gamer025
fix: Ingame ore silo logs should now log deposit actions correctly
config: New config for silo logs
/:cl:

### CODE REVIEW

File: code/__DEFINES/logging.dm [hunk 1]
```diff
@@ -40,6 +40,7 @@
 #define LOG_VICTIM (1 << 19)
 #define LOG_RADIO_EMOTE (1 << 20)
 #define LOG_SPEECH_INDICATORS (1 << 21)
+#define LOG_SILO (1 << 22)
```
Comment (Mothblocks) [inline code/__DEFINES/logging.dm:43]: I don't think we need more of these. We're running out of room

File: code/__HELPERS/logging/silo.dm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/proc/log_silo(text)
+	if (CONFIG_GET(flag/log_silo))
```
Comment (Mothblocks) [inline code/__HELPERS/logging/silo.dm:2]: We don't need a config for every single logging entry

**Review by Mothblocks — COMMENTED** [review]
@ZephyrTFA Please advise because this feels like the type of thing that should be JSON logged if we got that merged

File: code/__HELPERS/logging/silo.dm (hunk 2 above)
Comment (Gamer025) [reply code/__HELPERS/logging/silo.dm]: Made it so that silo logs are always created

**Comment by ZephyrTFA** [comment]
Yeah, please convert this into using the json logging system.
See:
`code\modules\logging\`

You need to add a category, and thats about it really.

File: code/__HELPERS/logging/silo.dm [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/proc/log_silo(text)
```
[the remark this answers is not in the document]
Comment (Gamer025) [reply code/__HELPERS/logging/silo.dm]: Proc no longer exists

**Comment by Gamer025** [comment]
Reworked it to use the new JSON based system, PR should be good now

**Review by ZephyrTFA — APPROVED** [review]

**Comment by Mothblocks** [comment]
@ZephyrTFA Doesn't look like there's any version field for the ore silo? All new logs should have a version number set, right? Otherwise why do we have it?

**Comment by ZephyrTFA** [comment]
> @ZephyrTFA Doesn't look like there's any version field for the ore silo? All new logs should have a version number set, right? Otherwise why do we have it? The default is 0.0.0 if I remember right, which implies unstable.

Semver isn't a thing yet. You're thinking of the general logging rewrite still waiting on json encode to be fixed

**Comment by Mothblocks** [comment]
:( okay

**Comment by Mothblocks** [comment]
(also json_encode has been fixed for a long time)

**Review by ArcaneMusic — APPROVED** [review]

**Pull request merged**